import_export/admin.py

```diff
@@ -498,52 +501,68 @@ def import_action(self, request, *args, **kwargs):
             if not input_format.is_binary():
                 input_format.encoding = self.from_encoding
             import_file = import_form.cleaned_data['import_file']
-            # first always write the uploaded file to disk as it may be a
-            # memory file or else based on settings upload handlers
-            tmp_storage = self.write_to_tmp_storage(import_file, input_format)
-            # allows get_confirm_form_initial() to include both the
-            # original and saved file names from form.cleaned_data
-            import_file.tmp_storage_name = tmp_storage.name
-
-            try:
-                # then read the file, using the proper format-specific mode
-                # warning, big files may exceed memory
-                data = tmp_storage.read()
-                dataset = input_format.create_dataset(data)
-            except Exception as e:
-                import_form.add_error('import_file',
-                               _(f"'{type(e).__name__}' encountered while trying to read file. "
-                                 "Ensure you have chosen the correct format for the file. "
-                                 f"{str(e)}"))
-
-            if not import_form.errors:
-                # prepare kwargs for import data, if needed
-                res_kwargs = self.get_import_resource_kwargs(request, form=import_form, *args, **kwargs)
-                resource = self.choose_import_resource_class(import_form)(**res_kwargs)
-                resources = [resource]
-
-                # prepare additional kwargs for import_data, if needed
-                imp_kwargs = self.get_import_data_kwargs(request, form=import_form, *args, **kwargs)
-                result = resource.import_data(dataset, dry_run=True,
-                                              raise_errors=False,
-                                              file_name=import_file.name,
-                                              user=request.user,
-                                              **imp_kwargs)
-
-                context['result'] = result
 
+            if getattr(settings, "IMPORT_EXPORT_SKIP_ADMIN_CONFIRM", False):
```

Comment:
Do you think this should be settable on an individual resource? Probably fine for now, but this feels like something where granularity is desirable long term.

Reply:
Yes good idea.  I think it is fine for now and we can add it in later if required.